### ISSUE TITLE
Multi-site locale in asset path => file not found

### ISSUE THREAD

**Issue by Pfeifakopf** [statamic/cms#1304]
My sites config:
```
    'sites' => 
    [
        'de' => [
            'name' => config('app.name') . ' - DE',
            'locale' => 'de_DE',
            'url' => '/de/',
        ],
        'en' => [
            'name' => config('app.name') . ' - EN',
            'locale' => 'en_US',
            'url' => '/en/',
        ]
    ]
```
The asset path contains now the locale and the file could not be found.
```
https://domain.tld/de/path/to/file.jpg
```

**Comment by jasonvarga** [comment]
Asset path where? Are you using a tag? Seeing something in the CP?

Please provide some more details.

**Comment by Pfeifakopf** [comment]
The image is shown in the cp. The image is shown correctly, if I remove the local from path (/de). So I think, it's an mod_rewrite problem. For the assets the locale should not be added to the image path.

**Comment by jasonvarga** [comment]
Got it, thanks.

**Comment by Pfeifakopf** [comment]
Do you have a workaround for it? Would really help!

**Comment by jasonvarga** [comment]
I think it's because the default site is a subdirectory (/de) and asset URLs are being generated relative to that. A workaround (not a great one) is to make your default site `/` instead of `/de`.

**Comment by Pfeifakopf** [comment]
Hmm, all my contents are saved inside the subdirectory, about 120 pages. I've tried something like this, but this didn't work.
```
{{ glide src="{ image|remove_left:/de }" width="1280" height="800" }}
```

**Comment by jasonvarga** [comment]
What are you expecting to be at the root though if both sites live in subdirectories?

**Comment by Pfeifakopf** [comment]
Sorry, i think my issue was unclear, I try to explain it better.
I have a multi-site like in my ticket described.  I've added an assets field for an image, using the default asset container. If I choose now an image and glide adds the locale to the image url. Without glide the url is correct and the image is shown.
Regular: /img/asset/image.jpg
Glide: /de/img/asset/image.jpg
So if I use glide, the image could not be found from the browser.

**Comment by jasonvarga** [comment]
I understand that, but put aside the image stuff for a second. What are you expecting to see when you visit the root?

**Comment by jasonvarga** [comment]
Note to self: 
https://github.com/statamic/cms/blob/master/src/Imaging/GlideUrlBuilder.php#L61
Removing the site url prepending "fixes" it - it doesn't prepend the site anymore, but it will need some more testing against multiple sites. Especially when using different domains.

**Comment by SKuipers** [comment]
> Note to self:
> https://github.com/statamic/cms/blob/master/src/Imaging/GlideUrlBuilder.php#L61
> Removing the site url prepending "fixes" it - it doesn't prepend the site anymore, but it will need some more testing against multiple sites. Especially when using different domains.

I had the same issue where a localized multi-site was returning the wrong url for image assets in the non-default site. Your GlideUrlBuilder.php fix worked well. What kind of testing needs done to see this fix make it into the beta? Thanks!!

**Comment by jonassiewertsen** [comment]
I did use the following workaround:
https://github.com/statamic/cms/pull/1842

It seems, that it would be a cleaner solution to change the path here:
https://github.com/statamic/cms/blob/master/src/Imaging/GlideUrlBuilder.php#L61

**Comment by jasonvarga** [comment]
Fixed by #2379
